From: Eric Sandeen <sandeen@redhat.com>
Date: Tue, 18 Dec 2007 12:09:47 -0600
Subject: [fs] ecryptfs: backport generic_file_aio_read
Message-id: 47680CEB.9090907@redhat.com
O-Subject: [RHEL 5.2 PATCH] 5/15: eCryptfs: backport generic_file_aio_read
Bugzilla: 228341

Backport to RHEL5 generic_file_aio_read API

diff --git a/fs/ecryptfs/file.c b/fs/ecryptfs/file.c
index 94efb9d..f5adc61 100644
--- a/fs/ecryptfs/file.c
+++ b/fs/ecryptfs/file.c
@@ -43,15 +43,15 @@
  * The function to be used for directory reads is ecryptfs_read.
  */
 static ssize_t ecryptfs_read_update_atime(struct kiocb *iocb,
-				const struct iovec *iov,
-				unsigned long nr_segs, loff_t pos)
+				char __user *buf,
+				size_t count, loff_t pos)
 {
 	int rc;
 	struct dentry *lower_dentry;
 	struct vfsmount *lower_vfsmount;
 	struct file *file = iocb->ki_filp;
 
-	rc = generic_file_aio_read(iocb, iov, nr_segs, pos);
+	rc = generic_file_aio_read(iocb, buf, count, pos);
 	/*
 	 * Even though this is a async interface, we need to wait
 	 * for IO to finish to update atime
